ls1intum/Artemis
# Support automatic assessment of activity diagrams (Part 2/2)

### Checklist

### Motivation and Context


To further reduce the assessment effort of tutors in UML modeling exercises, Artemis should provide automatic assessments for every modeling exercise type. At the moment, the system only supports the automatic assessment of UML class diagrams.

### Description

We extend the automatic assessment component Compass by UML activity diagrams and refactor the relevant parts to facilitate further extensions by other diagram types in the future. The changes are separated in two parts:
* **Part 1**:  Extend the model within Compass by UML activity diagrams and refactor/improve the UML related classes. Additionally, extensive tests for **all** UML related classes are added.
* **Part 2**: Change the usage of the concrete UMLClassDiagram class with the abstract class UMLDiagram throughout the system and extend/refactor the JSON parser that is responsible for parsing the JSON representation of UML diagrams to also support activity diagrams. 

The refactoring contains:
* restructuring and simplifying the code
* extracting common/duplicate code
* improving/unifying variable naming
* adding JavaDoc

**Part 2 is done in this PR. Part 1 was done in https://github.com/ls1intum/Artemis/pull/808.**

### Steps for Testing


1. Log in to Artemis
2. Submit several activity diagrams
3. Assess activity diagrams
4. Notice automatic assessment of common parts of the diagrams

Note: The automatic assessment of UML class diagrams should work as before!
